fl4health/clients/basic_client.py

```diff
@@ -122,17 +123,21 @@ def get_parameters(self, config: Config) -> NDArrays:
     def set_parameters(self, parameters: NDArrays, config: Config) -> None:
         """
         Sets the local model parameters transfered from the server using a parameter exchanger to coordinate how
-        parameters are set. If it's the first time the model is being initialized, we assume the full model is being
-        initialized and use the FullParameterExchanger() to set all model weights.
+        parameters are set. If it's the first time the model is being initialized and the model weights have not been
+        initialized, we assume the full model is being initialized and use the FullParameterExchanger() to set all
+        model weights.
 
         Args:
             parameters (NDArrays): Parameters have information about model state to be added to the relevant client
                 model but may contain more information than that.
             config (Config): The config is sent by the FL server to allow for customization in the function if desired.
         """
         assert self.model is not None
-        if not self.model_weights_initialized:
-            self.initialize_all_model_weights(parameters, config)
+
+        if not self.first_parameters_set:
```

Comment:
I think there is a small edge case that we might not be thinking about here when the warm up module is only a subset of the weights being initialized. Consider a simple example, such as FedProx, but we only load the first layer of the warm up module into the FedProx model to be trained. The rest of the FedProx model will be initialized differently per client (unless the seeds are well set etc.), which isn't what we want. 

One way we could get around this is to initialize the model with the server weights first, then load the warm up model to replace the specified layers. We could have the `BasicClient` accept an optional `WarmedUpModule` and if it's there, we perform a load with the class. This method would then be something like
```
if not self.model_weights_initialized:	
    self.initialize_all_model_weights(parameters, config)
    if self.warmed_up_module:
        self.warmed_up_module.load_from_pretrained(self.model)
...
```
We can talk about this a bit more if you'd like. This would mean we wouldn't need `first_parameters_set` anymore.